### PR TITLE
[WFENG-2848] autologin with avoid root redirect

### PR DESCRIPTION
**Pull request by bgerrity**
## Summary

Uses local storage rather than session storage to hold the state value for redirects after auth. This is a relaxation.

The `returnUrl` parameter no longer made sense when the login page was disused; the safe redirect to origin was being hit during an auto token refresh. ~~The previous behavior could cause redirects to root due to a race condition between the automatic token refresh and calls to the server.~~ That lead to user complaints.

Moves some redirect logic into the root layout.

With guidance from security, using a TTL behavior for the state redirects to address replay concerns.

## Test Plan

Deployed to gizmo for both domains and trad.

fake expired token for testing stale token, ddbuild for legacy envs
```
localStorage.setItem("AuthUser", '{"idToken":"eyJhbGciOiJSUzI1NiIsImtpZCI6IjY3YWEyNmIxLTMzODQtZGEzMi04M2E4LWIyOGMwMDU4MDI3YyJ9.eyJhdWQiOiAiYXRsYXMtc3RhZ2luZyIsICJkYXRhY2VudGVyIjogInVzMS5kZGJ1aWxkLmlvIiwgImVtYWlsIjogImJyZW5kYW4uZ2Vycml0eUBkYXRhZG9naHEuY29tIiwgImV4cCI6IDEwMCwgImdyb3VwcyI6IFtdfQo.aGE8Z2oH8NKq9X_3ivuaXsc2J_WSh71tT4_HQZlEHDPBP87uX6lYVITjLTCQSYpA5sABqmFCLHpL6VAMC9B53cFk4j5D9Gxg7pFfnVbPfafSg4iuaoB4AzAd_Gyyzf2cEzxqWcl1oaxTti69EcsjE-BronbnY74k8sGMsegKjqormsHg4QsuzOOY8q2ZPtU1pYuslCY0QeBbtfBb9vInsSocopREptl20q37e3QRGRV0031mn2ZR4qW0zM7YD1IstbUDWJioDAl2cpJL-VoUkqrRCCfUbNFBaKcQHurJMRCopRcK999499XSNR423VcbAFc9ezFim5al6uGBOHTrDw=","name":"brendan.gerrity","email":"brendan.gerrity@datadoghq.com"}');
```

fake expired token for testing stale token, us1.staging for domains
```
localStorage.setItem("AuthUser", '{"idToken":"eyJhbGciOiJSUzI1NiIsImtpZCI6IjY3YWEyNmIxLTMzODQtZGEzMi04M2E4LWIyOGMwMDU4MDI3YyJ9.eyJhdWQiOiJhdGxhcyIsImRhdGFjZW50ZXIiOiJ1czEuc3RhZ2luZy5kb2ciLCJlbWFpbCI6ImJyZW5kYW4uZ2Vycml0eUBkYXRhZG9naHEuY29tIiwiZXhwIjoxMDAsImdyb3VwcyI6W119.aGE8Z2oH8NKq9X_3ivuaXsc2J_WSh71tT4_HQZlEHDPBP87uX6lYVITjLTCQSYpA5sABqmFCLHpL6VAMC9B53cFk4j5D9Gxg7pFfnVbPfafSg4iuaoB4AzAd_Gyyzf2cEzxqWcl1oaxTti69EcsjE-BronbnY74k8sGMsegKjqormsHg4QsuzOOY8q2ZPtU1pYuslCY0QeBbtfBb9vInsSocopREptl20q37e3QRGRV0031mn2ZR4qW0zM7YD1IstbUDWJioDAl2cpJL-VoUkqrRCCfUbNFBaKcQHurJMRCopRcK999499XSNR423VcbAFc9ezFim5al6uGBOHTrDw=","name":"brendan.gerrity","email":"brendan.gerrity@datadoghq.com"}');
```

## Release Plan

As previously, this is the first of three PRs to effect the change into dd-source.

### CODE REVIEW

**Comment by bgerrity** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/temporalio-ui/10?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10** <a href="https://app.graphite.dev/github/pr/DataDog/temporalio-ui/10?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9** <a href="https://app.graphite.dev/github/pr/DataDog/temporalio-ui/9?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bgerrity and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: vite.config.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import { uiServerPlugin } from './plugins/vite-plugin-ui-server';
 
 export default defineConfig({
   plugins: [sveltekit(), temporalServer(), uiServerPlugin()],
+  build: {
+    minify: false,
+  },
```
Comment (sonalparab) [inline vite.config.ts:13]: I think we should remove this because we don't want users to see it?

File: src/lib/utilities/route-for.ts [hunk 2]
```diff
@@ -253,16 +254,16 @@ const routeForImplicitFlow = (
...
   authorizationUrl.searchParams.set('nonce', nonce);
 
   // state stores a reference to the redirect path
   const state = crypto.randomUUID();
-  sessionStorage.setItem(
-    state,
-    currentSearchParams.get('returnUrl') ?? (originUrl || '/'),
-  );
+  const stateUrl =
+    currentSearchParams.get('returnUrl') ?? window.location.href ?? '/';
```
Comment (bgerrity) [inline src/lib/utilities/route-for.ts:265]: The `window.location.href` change is the main item for the PR.

**Review by sonalparab — APPROVED** [review]
Works for me, thanks for fixing this

**Pull request merged**